### ISSUE TITLE
Allow continuation of subsequent steps if step fails in yourney

### ISSUE THREAD

**Issue by amergasevicsbsit** [elastic/synthetics#850]
In my TypeScript (TS) project, I have implemented a set of validation steps to assess the responses of API calls on a web page. These steps also include monitoring the duration of execution, among other criteria. Typically, a journey comprises around 10 such steps, with only one step having a UI component (to pick up a auth token). This is because we were unable to use a heartbeat mechanism for regular lightweight monitoring.

In the event that one of the steps fails, such as the third or fourth, the remaining 5-6 steps are currently skipped. This behavior is not ideal for our testing needs.

I understand that Playwright, as well as tools like Elastic/Synthetics, do not provide a built-in option to continue with subsequent test cases in the event of a failure, unless the tests are wrapped in a catch block. However, I would prefer not to use this approach.

To address this situation, I propose implementing a flag in the configuration that would allow the tests to continue running even if a failure occurs. This would provide a behavior similar to what you find in Cypress, where a failure in one "it" block does not lead to the failure of subsequent test steps.

This feature would enhance the robustness of our test suite and allow us to investigate multiple failures in a single test run, helping us identify and address issues more efficiently.

Please consider this proposal for our TypeScript project, as it would greatly improve our testing workflow and result in more comprehensive test reports.

**Comment by paulb-elastic** [comment]
Hi @amergasevicsbsit, interesting use case, but not sure it fits in with our direction for Synthetics at this time.

The Synthetics journeys focus on a series of interactions that are needed to move from one step to another, which is why we've built the behaviour like this (for example, I need to log in before I can see my account details).  In such cases, it is not possible to run step _n+1_ if step _n_ failed.

Adding the ability to override doesn't typically fit into this approach, and also complicates when errors are created etc.

We would always suggest having separate monitors for each _thing_ they are testing, even in a typical browser interaction (i.e. not just your API use case).

Could you not split each of your checks into separate monitors so that they all run independently?

**Comment by amergasevicsbsit** [comment]
@paulb-elastic thanks for the quick Response. Not sure I understand "split into multiple monitors". So problem for using for example regular HTTP monitor is I need bearer token for most of the APIs I want to monitor, which I need to generate each time. Heartbeat token generated only works with couple of them.
So my use case is I want to mimic journeys that users take but not through UI. Since we use Microsoft Authenticator and Basic auth I need this 1 UI step.
Followed by that are series of API calls that "mimic" what would happen if user enters the page.

**Comment by amergasevicsbsit** [comment]
Does this require that each of these are in separatey journeys:
Journey 1:
Monitor 1
Step 1: get bearer token
Step 2: API call a
Journey 2
Monitor 2
Step 1: get bearer token
Step 2: API call b
....
Based on documentation one journey can use only 1 monitor?

**Comment by vigneshshanmugam** [comment]
Yes, Journey and Monitor are 1-1 relationship, When you push these project based monitors, each journey ends up creating a single monitor.
